DS13/code/modules/overmap/ai.dm

```diff
@@ -0,0 +1,114 @@
+	desc = "A dangerous ship which resembles a bird. It has a modest armament and is highly maneuverable."
+	icon = 'DS13/icons/overmap/warbird.dmi'
+	icon_state = "warbird"
+	main_overmap = FALSE
+	class = "enemy-romulan" //Feel free to add overmap controls for AIs later, future me.
+	damage_states = FALSE //Damage FX
+	damage = 10 //Will turn into 20 assuming weapons powered
+	AI_enabled = TRUE //Start with an AI by default
+	faction = "romulan" //Placeholder
+	var/datum/overmap_event/linked_event
+	turnspeed = 3
+
+/obj/structure/overmap/AI/explode()
+	if(linked_event)
+		linked_event.check_completion(src)
+	. = ..()
+
+/obj/structure/overmap/AI/freighter
+	name = "Federation Frigate"
+	desc = "A minimally armoured tug with strong shields. It has next to no offensive power."
+	icon = 'DS13/icons/overmap/freighter.dmi'
+	icon_state = "freighter"
+	main_overmap = FALSE
+	class = "starfleet-freighter" //Feel free to add overmap controls for AIs later, future me.
+	damage_states = FALSE
+	damage = 0
+	faction = "starfleet"
+	max_shield_health = 300
+
+
+/obj/structure/overmap/proc/take_control() //Take control of our ship, make it into an AI
+	START_PROCESSING(SSobj, src) //Need to process to check for targets and so on.
+	name = "[name] ([rand(0,1000)])"
+	shield_power = 2 //So theyre not ultra squishy
+	weapon_power = 1
+	engine_power = 4
+	max_shield_power = 4
+	max_weapon_power = 4
+	max_engine_power = 4
+	power_slots -= 4
+	AI_enabled = TRUE //Let the computer take the wheel
+	check_power()
+
+/obj/structure/overmap/proc/remove_control() //Stop the AI from controlling the ship
+	AI_enabled = FALSE
+	STOP_PROCESSING(SSobj,src)
+	name = initial(name)
+	nav_target = null
+	target = null
+
+/obj/structure/overmap/process()
+	if(!AI_enabled) //Process is ONLY called for ai ships. We don't want a non ai ship doing this or consequences
+		return
+	if(vel < max_speed)
+		vel += acceleration
+	if(!process) //Start process allows the ship to move. It is called at init but if for some reason it stops, we need to reboot it.
+		process = TRUE
+		start_process()
+	if(target) //We have a target locked in
+		if(get_dist(src, target) > range) //Target ran away. Move on.
+			if(force_target)
+				target = force_target
+				nav_target = force_target //If we have a force target, we're an actor in a mission and NEED to return to hunt down our quarray after shooting at the players.
+				return
+			target = null //Don't shoot them, but keep chasing them UNLESS we're being forced to chase another.
+			pick_target()
+		if(behaviour == "peaceful") //Peaceful means never retaliate, so return
+			return
+		nav_target = target
+		fire(target, damage)//Shoot the target. This can either be us shooting on aggressive mode, or us being hit by the attacker.
+		return //No need to pick another target if we have one and theyre in range
+	else
+		pick_target()
+
+/obj/structure/overmap/proc/pick_target()
+	for(var/obj/structure/overmap/OM in orange(src, range))
```

Comment:
This feels inefficent. Can you do a better check by a list for all overmap structures in range? (And if none in range start moving in a certain manner? [Outside PR scope])